docs/design/reserve-action.md

```diff
@@ -0,0 +1,261 @@
+large resources can not be scheduled due to insufficient resources. In extreme cases, tasks with large resource requirement 
+can only be scheduled when all the small resource application pods are completed. This situation needs to be avoided as much
+as possible. When a task cannot be scheduled for a long time, a special scheduling mechanism need to be provided for it to
+ensure that the task can be scheduled as soon as possible.
+
+![pod-starving](../../docs/images/pod-starving.png)
+
+In `Time0`, there are seven pods to be scheduled with one node whose allocated resources are `3C3G` in cluster. The `pod-0` has
+the highest priority in the unscheduled pod queue and will be prioritized to be schedule. In `Time1`, `pod-1` `pod-2` `pod-3` 
+were bound to the node and begin to run. The `pod-4` has the highest priority in the unscheduled pod queue and will be 
+prioritized to be schedule. In `Time2`, `pod-1` was completed, and `1C1G` resource was released from node. The scheduler
+try to schedule `pod-4`, but idle resource of node can not satisfy pod's requirement. The `pod-4` can not be scheduled and 
+was skipped. The schedule try to schedule `pod-5` and succeed in `Time3`. In `Time4`, `pod-2` and `pod-3` were `completed`
+and `2C2G` was released from node. Because of that the idle resource of node can not satisfy requirement of pod `pod-4`, `pod-4`
+was skipped. The `pod-6` and `pod-7` were bound to node and begin to run. In `Time6` all pods except pod `pod-4` were completed
+and all resources of node were release. The scheduling condition of pod `pod-4` is satisfied, `pod-4` was bound to node and 
+begin to run. In the above scenario, the `pod-4` can not be scheduled in time because a large number of small resource requirement
+pods occupied the node resources.
+ 
+In order to solve such problems, the reserve action is introduced into the scheduling policy. The `reserve` action can 
+prevents the large resource requirement task from being starved by reserving node resources for it and not allowing the other 
+tasks to be scheduled on the node.
+
+![reserve-resource-for-pod](../../docs/images/reserve-resource-for-pod.png)
+
+Different from the original scheduling logic, in `Time4`, although there are `2C2G` idle resources on the node, it does not
+schedule new tasks to the nodes, but reserves these resources for `pod-4`. In `Time5` `pod-6` and `pod-7` remain `Pending`. 
+In `Time 6`, `pod-5` completed and all resources on node were released. The scheduling condition of `pod-4` is satisfied, `pod-4`
+is bound to the node and starts `Running`.
+
+## Function Detail 
+
+### Explanation of terms
+
+#### Job Ready
+
+If the running condition of a job is satisfied, then the job is in the `Job Ready` state, and after scheduling, the job can be in the 
+`Job Running` state. The running condition of a job may include restrictions such as `gang-scheduler` and so on.
+
+#### Job Condition Ready
```

Comment:
It is defined, but never used in this PR?

Reply:
This will be used to judge whether the job need to be reserved resources. If there are already nodes that have reserved resources for this job, there is no need to perform a `reserve` operation for them.

Reply:
But it is not used in other places in this PR?

Reply:
there is a quote in `Job starving`
···
#### Job starving

The starving job needs to meet the following conditions

* The creation time of job is greater than the set starving job threshold time
* Job state is not `Job Ready`
* Job state is not `Job Condition Ready`

···

Reply:
Sorry, just missed it.